src/js/actions/guides.js

```diff
@@ -0,0 +1,329 @@
+            createObj = documentLib.insertGuide(docRef, orientation, position);
+
+        return descriptor.playObject(createObj)
+            .then(function () {
+                var eventKind = adapterOS.eventKind.LEFT_MOUSE_DOWN,
+                    coordinates = [x, y];
+                        
+                return adapterOS.postEvent({ eventKind: eventKind, location: coordinates });
+            });
+    };
+    createGuideAndTrack.reads = [locks.JS_UI];
+    createGuideAndTrack.writes = [locks.JS_DOC];
+
+    /**
+     * Updates the given guide's position or creates a new guide if necessary
+     *
+     * @param {object} payload
+     * @param {number} payload.documentID Owner document ID
+     * @param {number} payload.index Index of the edited guide
+     * @param {string} payload.orientation New orientation of the guide
+     * @param {number} payload.position New position of the guide
+     *
+     * @return {Promise}
+     */
+    var setGuide = function (payload) {
+        return this.dispatchAsync(events.document.history.nonOptimistic.GUIDE_SET, payload)
+            .bind(this)
+            .then(function () {
+                return this.transfer(resetGuidePolicies);
+            });
+    };
+    setGuide.reads = [];
+    setGuide.writes = [locks.JS_DOC];
+    setGuide.transfers = [resetGuidePolicies];
+
+    /**
+     * Deletes the given guide
+     *
+     * @param {object} payload
+     * @param {number} payload.documentID Owner document ID
+     * @param {number} payload.index Index of the edited guide
+     *
+     * @return {Promise}
+     */
+    var deleteGuide = function (payload) {
+        return this.dispatchAsync(events.document.history.nonOptimistic.GUIDE_DELETED, payload)
+            .bind(this)
+            .then(function () {
+                return this.transfer(resetGuidePolicies);
+            });
+    };
+    deleteGuide.reads = [];
+    deleteGuide.writes = [locks.JS_DOC];
+    deleteGuide.transfers = [resetGuidePolicies];
+
+    var queryCurrentGuides = function (document) {
+        var docRef = documentLib.referenceBy.id(document.id);
+
+        return _getGuidesForDocumentRef(docRef)
+            .bind(this)
+            .then(function (guides) {
+                var payload = {
+                    document: document,
+                    guides: guides
+                };
+
+                return this.dispatch(events.document.GUIDES_UPDATED, payload);
+            });
+    };
+    queryCurrentGuides.reads = [locks.PS_DOC];
+    queryCurrentGuides.writes = [locks.JS_DOC];
+
+    // Event handlers for guides
+    var _guideSetHandler = null,
+        _guideDeleteHandler = null;
+
+    /**
+     * Register event listeners for guide edit/delete events
+     * 
+     * @return {Promise}
+     */
+    var beforeStartup = function () {
+        // Listen for guide set events
+        _guideSetHandler = function (event) {
+            var target = event.null._ref;
```

Comment:
are you pretty sure `.null` will always be there?  If not, you could use objUtil.getPath(event, "null._ref")